.github/actions/spell-check/whitelist.txt

```diff
@@ -0,0 +1,3240 @@
+estmd
+estmetadata
+estpasses
+ests
+esult
+esultmacros
+ETB
+etfx
+ethanschoonover
+ETW
+ETX
+EUDC
+ev
+EVENTID
+eventing
+everytime
+evflags
+evt
+ewdelete
+ewtonsoft
+exe
+execd
+executables
+executionengine
+exemain
+EXETYPE
+exeuwp
+exewin
+exitwin
+expectedinput
+expr
+EXPUNGECOMMANDHISTORY
+EXSTYLE
+EXTENDEDEDITKEY
+EXTKEY
+Extralight
+EXTTEXTOUT
+Fabrikam
+facename
+FACENODE
+FACESIZE
+FAFAFA
+failfast
+FAILIFTHERE
+fallthrough
+FARPROC
+fba
+fbb
+fbd
+fbdc
+FBE
+fcb
+fcc
+fcfdfa
+fcharset
+fclose
+fcntl
+fd
+FDAB
+fdb
+fdc
+fdd
+fde
+FDF
+FDFD
+fdopen
+fdpi
+fdw
+fea
+FECAC
+fehlende
+fehlenden
+fehlt
+fesb
+ffb
+ffc
+FFCCCCCC
+FFDE
+FFF
+fffd
+FFFDu
+ffff
+ffffff
+FFFFFFFF
+FFFFFFFFFFFF
+FFrom
+ffu
+FGCOLOR
+fgetc
+fgetwc
+fgidx
+FILEDESCRIPTION
+fileno
+FILEPATH
+FILESUBTYPE
+FILESYSPATH
+filesystem
+FILETIME
+FILETYPE
+FILEW
+fililng
+FILLATTR
+FILLCONSOLEOUTPUT
+Filledbox
+FILTERONPASTE
+finalizer
+FINDCASE
+FINDDLG
+FINDDOWN
+finden
+findlocalename
+FINDSTR
+FINDSTRINGEXACT
+FINDUP
+Firefox
+FIter
+FIXEDCONVERTED
+Flg
+flyout
+fmodern
+fmtarg
+fmtid
+FOLDERID
+FONTCHANGE
+fontdlg
+FONTENUMDATA
+FONTENUMPROC
+FONTFACE
+FONTFAMILY
+FONTHEIGHT
+fontlist
+FONTOK
+FONTSIZE
+FONTSTRING
+fonttbl
+FONTTYPE
+FONTWEIGHT
+FONTWIDTH
+FONTWINDOW
+forceinline
+FORCEOFFFEEDBACK
+FORCEONFEEDBACK
+FORCEV
+foreach
+fprintf
+framebuffer
+FRAMECHANGED
+fre
+freopen
+frontend
+Fscreen
+FSCTL
+FSINFOCLASS
+fsproj
+fstream
+fte
+Ftm
+fullcolor
+fullscreen
+fullwidth
+func
+FUNCTIONCALL
+fuzzer
+fuzzmap
+fuzzwrapper
+fwdecl
+fwe
+fwlink
+fyi
+Gabriola
+Gatta
+gb
+gci
+gcx
+gcy
+gdi
+gdip
+gdirenderer
+GENERATEPROJECTPRIFILE
+geopol
+GETALIAS
+GETALIASES
+GETALIASESLENGTH
+GETALIASEXES
+GETALIASEXESLENGTH
+getboundingrectangles
+GETCARETWIDTH
+getch
+getchar
+GETCOMMANDHISTORY
+GETCOMMANDHISTORYLENGTH
+GETCONSOLEINPUT
+GETCONSOLEPROCESSLIST
+GETCONSOLEWINDOW
+GETCOUNT
+GETCP
+GETCURSEL
+GETCURSORINFO
+GETDISPLAYMODE
+GETDISPLAYSIZE
+GETDLGCODE
+GETDPISCALEDSIZE
+GETFONTINFO
+GETFONTSIZE
+getglyphs
+GETHARDWARESTATE
+GETHUNGAPPTIMEOUT
+GETICON
+GETITEMDATA
+GETKEYBOARDLAYOUTNAME
+GETKEYSTATE
+GETLARGESTWINDOWSIZE
+GETLBTEXT
+getline
+getlocalename
+getlocalenamelength
+GETMINMAXINFO
+GETMOUSEINFO
+GETNUMBEROFFONTS
+GETNUMBEROFINPUTEVENTS
+GETOBJECT
+GETPOS
+GETSELECTIONINFO
+getset
+getstring
+getstringlength
+GETTEXT
+GETTEXTLEN
+GETTITLE
+GETWAITTOKILLSERVICETIMEOUT
+GETWAITTOKILLTIMEOUT
+GETWHEELSCROLLCHARACTERS
+GETWHEELSCROLLCHARS
+GETWHEELSCROLLLINES
+getwriter
+Gfun
+gfx
+ggggg
+gggggg
+ggggggg
+gh
+github
+githubusercontent
+gle
+globals
+gmail
+GMEM
+GNUC
+Goldmine
+gonce
+Google
+goutput
+Gravell's
+grayscale
+GREENSCROLL
+Grehan
+Greyscale
+gridline
+Grie
+Griese
+groupbox
+gsl
+GTP
+guc
+gui
+guidatom
+guidgenerator
+GValue
+gw
+GWL
+GWLP
+gwsz
+HABCDEF
+halfwidth
+HALTCOND
+HANGEUL
+hardcoded
+hardcodes
+hashalg
+HASSTRINGS
+hbitmap
+hbm
+HBMMENU
+hbmp
+hbr
+hbrush
+hc
+HCmd
+hdc
+hdr
+HDROP
+hdrstop
+HEIGHTSCROLL
+Hernan
+herunterzuladen
```

Comment:
Is this a random German word? Where'd that come from?

Reply:
https://github.com/microsoft/terminal/blob/be77b2a2e98420db9d275d68f35b55ac58bc93c0/src/tools/U8U16Test/U8U16Test.vcxproj#L124
I didn't want to exclude the entire file (type) just because it had a small bit of German.

I think a file for german whitelisted tokens would be a reasonable thing.

While it's technically possible to give each file its own whitelist, I've always imagined that would make the runtime more expensive. (The current model runs all files together and then the output from all files together again -- otherwise the arrays have trouble sorting for very large repositories.)

I'd have to really think through how to handle this. I think the easiest approach would be for me to force people to run the spell checker multiple times w/ distinct exclusions. -- So one pass might use .github/actions/spell-check/{whitelist,excludes}.txt and another pass .github/actions/spell-check-foreign/{whitelist,excludes}.txt -- where the excludes in the first case excludes the files w/ troublesome text and the excludes in the second case excludes everything that isn't troublesome.

Happy to talk out other approaches.

Reply:
Actually..... we would consider that block to be wrong. It should have had the English error text there. I bet @german-one created that `.vcxproj` when working on the u8u16 converters and VS inserted the text in the bottom in German per his locale and we didn't notice. It's just our practice to have everything be in English as that's our company's working language. (Except where necessary like with some example German text for a test scenario or something...)

Reply:
Right. I've worked in multinational corporations. I don't personally speak German and wasn't in a good enough position to know what to do with it.

I'm happy to let someone remove that text and thus have the word drop from the whitelist.

Reply:
Yeah, that was me. But since it was automatically generated text I didn't even notice it. Do you want me to start a PR with an English translation of this message?

FWIW I can ensure that I'll refrain from German in conversations and code comments (instead I'll stick to some kind of English which feeds the spell checker). But I can't guarantee that Microsoft's products will keep trying 😛

Reply:
Personally, I'd want you to create such a PR. I'm not in any position to fix your German.

Once this lands, it should be harder for those products to sneak things like this in. :-)

Reply:
No problem, the link in the message points to the docs where the English text can be found.
Feel free to review #4838.

Reply:
Thanks @german-one.